Add more details to cross-cluster page #20230811 #20231102 #20230804

AntonEliatra opened the pull request:
### Description
_Added more details to cross cluster page focusing on bare metal, kubernetes/helm and curl using UI dev tools_

### Issues Resolved
_#20230811 #20231102 #20230804 from list on issue #4314_


### Checklist
- [x ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

hdhalter commented:
Thanks, Anton! Can you please resolve the Vale errors in the file before we send it for technical review?

AntonEliatra commented:
@hdhalter all errors have now been resolved, please let me know if there are other changes needed

hdhalter commented on _security/access-control/cross-cluster-search.md:
````diff
@@ -167,6 +167,9 @@ curl -k -XPUT -H 'Content-Type: application/json' -u 'admin:<custom-admin-passwo
 }'
 ```
+All of the curl requests can also be sent using OpenSearch Dashboards Dev tools
+{: .label .label-green }
````
```suggestion
{: .tip }
```
@AntonEliatra - You'll want to use [Callouts](https://github.com/opensearch-project/documentation-website/blob/main/FORMATTING_GUIDE.md#callouts) for this type of info, not [Labels](https://github.com/opensearch-project/documentation-website/blob/main/FORMATTING_GUIDE.md#labels).

hdhalter replied:
Can we add another H2 above this section? Something like, "Configuring cross-cluster search from Dashboards"? (I would rethink the other headings, too.)

AntonEliatra replied:
@hdhalter I dont think we discussed this today, can you elaborated please

AntonEliatra replied:
There is only one screenshot just to demonstrate that all of the curl commands can be executed from dev tools, Do you mean to add more screenshots and expand on this?

hdhalter commented on _security/access-control/cross-cluster-search.md:
````diff
@@ -243,3 +246,56 @@ curl -XGET -k -u booksuser:password 'https://localhost:9250/opensearch-ccs-clust
+}'
+```
+It is enough to point to only one of the node IPs on the remote cluster, as node discovery process will query all nodes in that cluster.
+{: .label .label-green }
````
```suggestion
{: .tip }
```

hdhalter reviewed (changes requested):
Thanks, @AntonEliatra! I have a few comments, but will send for a tech review.

hdhalter commented:
Hi @scrawfor99, can you please review for technical accuracy? Thanks!

scrawfor99 reviewed (approved):
Thanks @AntonEliatra. Everything written looks correct on review. 

I do have a couple general points for you and @hdhalter to consider:

1. I am not sure why we are splitting the examples into different infrastructure types. Looking at the [examples](https://github.com/opensearch-project/documentation-website/pull/6657/files#diff-217a365a33a8bde963770ca53002d9585a20c7773d8da24766499b554f62ba39R250), these seem standard across platforms. I am not sure whether splitting these into supported frameworks would cause a misunderstanding or perception that some of the commands only work on one system. I would expect all of the standard requests and commands to properly work on all systems. 

2. I am not sure whether this page is better served in the search documentation directory. Other than the fact that you will need to authorize and authenticate with the requests (which is true for all of the requests when security is enabled) I don't see any security specific.

AntonEliatra commented:
@scrawfor99 thanks for above feedback, I do agree that the docker example is quite comprehensive and commands apply to any installation, however there has been quite a few questions as to why its docker specific and how to configure this via UI, we could just state at the very top, that the example is using docker, but the steps/commands apply to any install, but will this sufficiently address the existing questions? @hdhalter what do you think?

Regarding point 2, I do agree, this page seems better placed under "Search" docs, I'm happy enough to change.

hdhalter commented:
@peterzhuamazon - Can you please comment on Anton's suggestion above for positioning the Docker instructions?

scrawfor99 commented:
Hi @AntonEliatra, gotcha. I definitely don't want to be a pain or anything. I think having the documentation like you wrote it would be just fine. My comment was more for the sake of @Naarcha-AWS or @hdhalter etc. for when they review as something they may consider.

I am in a bit of a unique case since I help with the documentation a lot but don't have as much opinion on the formatting as the technical content etc. Since you nailed the details there I have no issues :) 

Looks like Peter may be able to give some advice on where things can best go

hdhalter commented:
I think it's a really good call-out, @scrawfor99. I'm trying to get a second opinion from someone who has more technical knowledge of these permissions. Peter suggested @bbarani. 

I also found another topic called '[Cross cluster search](https://opensearch.org/docs/latest/observing-your-data/cross-cluster-replication/)' in the Observability section. And it doesn't point to where to set up permissions. Instead, it points to the Observability repo. We should rename or consolidate the topics.

hdhalter commented:
How about we keep the content in this topic generic and move anything docker-specific to https://opensearch.org/docs/latest/install-and-configure/install-opensearch/docker/?

AntonEliatra commented:
@hdhalter I moved the page to search section, I also had a look at the link that was in Observability category ande noticed that its primarily for PPL, therefore I added it to PPL page

hdhalter commented:
Thanks, @AntonEliatra ! 

@Naarcha-AWS - I made a few more tweaks to the content to add some context, and fixed a few formatting issues. Please review at your convenience. Thanks!

Naarcha-AWS commented on _search-plugins/cross-cluster-search.md:
````diff
@@ -243,3 +243,56 @@ curl -XGET -k -u booksuser:password 'https://localhost:9250/opensearch-ccs-clust
 }
 ```
+
+## Sample Bare metal/virtual machines setup
````
```suggestion
## Sample bare metal/virtual machines setup
```

natebower commented on _search-plugins/cross-cluster-search.md:
```diff
@@ -23,27 +22,25 @@ Cross-cluster search is exactly what it sounds like: it lets any node in a clust
 ## Authentication flow
 
-When accessing a *remote cluster* from a *coordinating cluster* using cross-cluster search:
+The following sequence describes the authentication flow when using cross-cluster search to access a *remote cluster* from a *coordinating cluster*. You can have different authentication and authorization configurations on the remote and coordinating cluster, but we recommend using the same settings on both.
```
```suggestion
The following sequence describes the authentication flow when using cross-cluster search to access a *remote cluster* from a *coordinating cluster*. You can have different authentication and authorization configurations on the remote and coordinating clusters, but we recommend using the same settings on both.
```

natebower commented on _search-plugins/cross-cluster-search.md:
```diff
@@ -23,27 +22,25 @@ Cross-cluster search is exactly what it sounds like: it lets any node in a clust
-## Permissions
-
-To query indexes on remote clusters, users need to have `READ` or `SEARCH` permissions. Furthermore, when the search request includes the query parameter `ccs_minimize_roundtrips=false` – which tells OpenSearch not to minimize outgoing and ingoing requests to remote clusters – users need to have the following additional permission for the index:
+To query indexes on remote clusters, users must have `READ` or `SEARCH` permissions. Furthermore, when the search request includes the query parameter `ccs_minimize_roundtrips=false` – which tells OpenSearch not to minimize outgoing and incoming requests to remote clusters – users need to have the following additional permission for the index:
```
```suggestion
To query indexes on remote clusters, users must have `READ` or `SEARCH` permissions. Furthermore, when the search request includes the query parameter `ccs_minimize_roundtrips=false`---which tells OpenSearch not to minimize outgoing and incoming requests to remote clusters---users need to have the following additional index permission:
```

natebower commented on _search-plugins/cross-cluster-search.md:
```diff
@@ -57,14 +54,14 @@ humanresources:
+## Sample Docker setup
 
-Save this file as `docker-compose.yml` and run `docker-compose up` to start two single-node clusters on the same network:
+To define permissions on Docker, save the following sample file as `docker-compose.yml`, and run `docker-compose up` to start two single-node clusters on the same network.
```
```suggestion
To define Docker permissions, save the following sample file as `docker-compose.yml` and run `docker-compose up` to start two single-node clusters on the same network:
```

natebower commented on _search-plugins/cross-cluster-search.md:
````diff
@@ -119,7 +116,7 @@ networks:
 ```
 
-After the clusters start, verify the names of each:
+After the clusters start, verify the names of each using the following commands.
````
Should this end in a colon?

AntonEliatra replied:
Thats changed now

natebower commented on _search-plugins/cross-cluster-search.md:
````diff
@@ -167,6 +164,9 @@ curl -k -XPUT -H 'Content-Type: application/json' -u 'admin:<custom-admin-passwo
   }
 }'
 ```
+All of the curl requests can also be sent using OpenSearch Dashboards Dev Tools.
````
```suggestion
All of the cURL requests can also be sent using OpenSearch Dashboards Dev Tools.
```

natebower replied:
"using the OpenSearch Dashboards Dev Tools console"?

natebower commented on _search-plugins/cross-cluster-search.md:
````diff
@@ -167,6 +164,9 @@ curl -k -XPUT -H 'Content-Type: application/json' -u 'admin:<custom-admin-passwo
 ```
+All of the curl requests can also be sent using OpenSearch Dashboards Dev Tools.
+{: .tip }
+![OpenSearch Dashboards UI for configuring remote cluster for Cross-cluster search]({{site.url}}{{site.baseurl}}/images/ccs-devtools.png)
````
Please reference in the preceding text.

AntonEliatra replied:
I've now added a line before the image, If thats what you meant @natebower

natebower replied:
@AntonEliatra Yes, correct. See my edits.

natebower commented on _search-plugins/cross-cluster-search.md:
````diff
@@ -224,7 +224,7 @@ curl -XPUT -k -u 'admin:<custom-admin-password>' -H 'Content-Type: application/j
 ```
 
-Both clusters must have the user, but only the remote cluster needs the role and mapping; in this case, the coordinating cluster handles authentication (i.e. "Does this request include valid user credentials?"), and the remote cluster handles authorization (i.e. "Can this user access this data?").
+Both clusters must have the user role present, but only the remote cluster needs the role and mapping. In this case, the coordinating cluster handles authentication (that is, "Does this request include valid user credentials?"), and the remote cluster handles authorization (that is, "Can this user access this data?").
````
```suggestion
Both clusters must have the user role, but only the remote cluster needs both the role and mapping. In this case, the coordinating cluster handles authentication (that is, "Does this request include valid user credentials?"), and the remote cluster handles authorization (that is, "Can this user access this data?").
```

natebower commented on _search-plugins/cross-cluster-search.md:
````diff
@@ -243,3 +243,56 @@ curl -XGET -k -u booksuser:password 'https://localhost:9250/opensearch-ccs-clust
 }
 ```
+
+## Sample bare metal/virtual machines setup
````
```suggestion
## Sample bare metal/virtual machine setup
```

natebower commented on _search-plugins/cross-cluster-search.md:
```diff
@@ -243,3 +243,56 @@ curl -XGET -k -u booksuser:password 'https://localhost:9250/opensearch-ccs-clust
+## Sample bare metal/virtual machines setup
+
+If you are running OpenSearch on a bare metal server or using a virtual machine, you can run the same commands, specifying the IP (or domain) of the OpenSearch cluster.
+For example, in order to configure remote cluster for Cross-cluster search, find the IP of the remote node or domain of the remote cluster and run command:
```
Either combine with the previous sentence or insert a line break. Should we tell/link the reader where they can find the IP or the domain?

natebower replied:
```suggestion
For example, in order to configure a remote cluster for cross-cluster search, find the IP of the remote node or domain of the remote cluster and run the following command:
```

natebower commented on _search-plugins/cross-cluster-search.md:
````diff
@@ -243,3 +243,56 @@ curl -XGET -k -u booksuser:password 'https://localhost:9250/opensearch-ccs-clust
+  }
+}'
+```
+It is enough to point to only one of the node IPs on the remote cluster, as node discovery process will query all nodes in that cluster.
````
```suggestion
It is sufficient to point to only one of the node IPs on the remote cluster because all nodes in the cluster will be queried as part of the node discovery process.
```

natebower commented on _search-plugins/cross-cluster-search.md:
```diff
@@ -243,3 +243,56 @@ curl -XGET -k -u booksuser:password 'https://localhost:9250/opensearch-ccs-clust
+It is enough to point to only one of the node IPs on the remote cluster, as node discovery process will query all nodes in that cluster.
+{: .tip }
+
+You can now run queries across both clusters.
```
```suggestion
You can now run queries across both clusters:
```

natebower commented on _search-plugins/cross-cluster-search.md:
````diff
@@ -243,3 +243,56 @@ curl -XGET -k -u booksuser:password 'https://localhost:9250/opensearch-ccs-clust
+```
+
+## Sample Kubernetes/Helm setup
+If you are using Kubernetes clusters to deploy OpenSearch, you have to configure the remote cluster using the `LoadBalancer` or `Ingress` services. The services that get created using the following [Helm]({{site.url}}{{site.baseurl}}/install-and-configure/install-opensearch/helm/) example are of the `ClusterIP` type and are only be accessible inside the cluster, therefore external services need t ...
````
Second sentence: Given that we already refer to services in both the first and third sentences, should we use a different word in "The services created"?

AntonEliatra replied:
I've reworded this slightly to:
```
## Sample Kubernetes/Helm setup
If you are using Kubernetes clusters to deploy OpenSearch, you need to configure the remote cluster using either the `LoadBalancer` or `Ingress`. The kubernetes services created using the following [Helm]({{site.url}}{{site.baseurl}}/install-and-configure/install-opensearch/helm/) example are of the `ClusterIP` type and are only accessible from within the cluster; therefore, you must use an externally accessible endpoint:
```

natebower replied:
@AntonEliatra Looks good, but the second instance of "Kubernetes" should be capitalized.

AntonEliatra replied:
@natebower thats fixed now

natebower reviewed (commented):
@AntonEliatra @Naarcha-AWS Please see my comments and changes and let me know if you have any questions. Thanks!

natebower commented on _search-plugins/cross-cluster-search.md:
````diff
@@ -167,6 +164,10 @@ curl -k -XPUT -H 'Content-Type: application/json' -u 'admin:<custom-admin-passwo
 ```
+All of the cURL requests can also be sent using OpenSearch Dashboards Dev Tools.
+{: .tip }
+Following image is the example of cURL request using Dev Tools:
````
```suggestion
The following image shows an example of a cURL request using Dev Tools:
```
